src/arch/x86_64/CodeGen.zig

```diff
@@ -1142,7 +1141,14 @@ fn allocMemPtr(self: *Self, inst: Air.Inst.Index) !u32 {
 }
 
 fn allocRegOrMem(self: *Self, inst: Air.Inst.Index, reg_ok: bool) !MCValue {
-    const elem_ty = self.air.typeOfIndex(inst);
+    return self.allocRegOrMemAdvanced(self.air.typeOfIndex(inst), inst, reg_ok);
+}
+
+fn allocTempRegOrMem(self: *Self, elem_ty: Type, reg_ok: bool) !MCValue {
+    return self.allocRegOrMemAdvanced(elem_ty, null, reg_ok);
```

Comment:
That's a good idea but we need to remember to implement stack GC at some point so that we don't run out of stack space.

Reply:
Can you clarify whether all stack allocations already needed GC or only the ones created by this new function because I missed doing something.  The idea is that not every needed allocation happens to match the type of an instruction.  In any case, so far I'm only using this temporarily until `fn air*Ptr` are rewritten to not load the entire value, but it could also be needed for certain some operations in the future.

Reply:
I expect all stack allocations needing GC, yes.